src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/SupplierSectionController.cs

```diff
@@ -50,18 +51,58 @@ public async Task<ActionResult<SupplierModel>> GetAsync(string orderId)
         }
 
         [HttpPut]
-        public ActionResult Update([FromRoute] [Required] string orderId, [FromBody] [Required] SupplierModel model)
+        public async Task<ActionResult> UpdateAsync(string orderId, SupplierModel model)
         {
-            if (orderId is null)
+            if (model is null)
+            {
+                throw new ArgumentNullException(nameof(model));
+            }
+
+            var order = await _orderRepository.GetOrderByIdAsync(orderId);
+
+            if (order is null)
             {
                 return NotFound();
             }
 
-            if (model is null)
+            var primaryOrganisationId = User.GetPrimaryOrganisationId();
+            if (primaryOrganisationId != order.OrganisationId)
             {
-                throw new ArgumentNullException(nameof(model));
+                return Forbid();
             }
 
+            order.SupplierId = model.SupplierId;
+            order.SupplierName = model.Name;
+
+            var address = model.Address;
+
+            if (order.SupplierAddress is null)
+                order.SupplierAddress = new Address();
+
+            order.SupplierAddress.Line1 = address.Line1;

```

Comment:
It'd be cool if we had a function on Address, or an extension method on Address, to do this for us, but isn't essential

Reply:
👍